### PR TITLE
mfa: reuse the same challenge for all U2F devices

### PR DESCRIPTION
**Pull request by awly**
Challenge is a random string that U2F devices must sign. The JS API
requires you to use the same challenge for all registered devices,
instead of one challenge per device that we had previously. See
https://fidoalliance.org/specs/fido-u2f-v1.0-nfc-bt-amendment-20150514/fido-u2f-javascript-api.html#dictionary-u2fsignrequest-members

Reuse the same challenge for U2F devices to match the JS API. Also,
propagate the version string to follow the spec exactly.

I will follow up with a `webassets` PR to support multiple devices.

### CODE REVIEW

**Comment by russjones** [comment]
@fspmarshall @a-palchikov Can you two review?

**Review by fspmarshall — APPROVED** [review]

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -24,6 +24,16 @@ import (
...
 	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gravitational/trace"
+	"github.com/gravitational/trace/trail"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
```
Comment (Joerger) [inline lib/auth/grpcserver.go:35]: Import block should go below teleport imports.

**Review by Joerger — APPROVED** [review]

File: lib/auth/auth.go [hunk 2]
```diff
@@ -2005,22 +2007,28 @@ func (a *Server) mfaAuthChallenge(ctx context.Context, user string, u2fStorage u
...
-				Dev:        dev,
-				Storage:    u2fStorage,
-				StorageKey: user,
-			})
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
+			u2fDevs = append(u2fDevs, dev)
+		default:
+			log.Warningf("skipping MFA device of unknown type %T", dev.Device)
```
Comment (a-palchikov) [inline lib/auth/auth.go:2012]:
```suggestion
			log.Warningf("Skipping MFA device of unknown type %T.", dev.Device)
```
more of a nit.

**Review by a-palchikov — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**